hibernate-core/src/main/java/org/hibernate/proxy/AbstractLazyInitializer.java

```diff
@@ -102,6 +104,19 @@ public final SharedSessionContractImplementor getSession() {
 		return session;
 	}
 
+	@Override
+	public final void setLockMode(LockMode lockMode) {
+		this.requestedLockMode = lockMode;
```

Comment:
Shouldn't this only be called when uninitialized?

```suggestion
		assert isUninitialized();
		this.requestedLockMode = lockMode;
```